tests/functional/object/mcg/test_mcg_performance_profiles.py

```
@@ -0,0 +1,478 @@
+        # Step 5: Verify endpoint pod count (min/max)
+        logger.info("Verifying endpoint pod count configuration")
+        current_count = len(endpoint_pods)
+        expected_min = spec["endpoint_count"]["min"]
+        expected_max = spec["endpoint_count"]["max"]
+
+        assert expected_min <= current_count <= expected_max, (
+            f"Endpoint pod count {current_count} not within expected range "
+            f"[{expected_min}, {expected_max}]"
+        )
+        logger.info(
+            f"Endpoint pod count: {current_count} (within range [{expected_min}, {expected_max}]) ✓"
```

Ruff (0.16.2)

[warning] Logging statement uses f-string

(G004)